### ISSUE TITLE
[Tweak] Recently opened files / file associations

### ISSUE THREAD

**Issue by Silent-Joker** [mRemoteNG/mRemoteNG#280]
Scratch this.... as i started to debug and delete and rename files its my fault for using the same name for connection files....

Maybe add a recently opened option for open connection files...

or i had it on the old site.

Make the connection files a special name like .mrcon that you can double click and it load mremote directly.

keep up the good work sorry

**Comment by sparerd** [comment]
I think there is something to your original report that we should investigate. Connections loading is not acting as expected when you cancel the password prompt box and then restart the program.

Replication steps:
1. Set a custom password on a connections file, save it.
1. Close and reopen mRemoteNG
1. Hit Cancel on the password prompt
1. Close and reopen mRemoteNG
1. You will not be prompted for a password (as expected). The root connection tree node will be erroneously named "PuTTY Saved Sessions".

Original bug report:
> I have used a password protected file that i created fresh with Beta 1. Today i clicked cancel when loading Mremote cause i didnt mean to load it. I than closed mremote. When i went to use it later on in the day it didnt open the password protected file. I tried to load all backup files and none are current. I looked everywhere and cannot find a recent file that i have protected.

> I tried to load the .xml files in notepad and all of the backup ones are viewable. The current one is not. It seems like its encrypted. So im not sure what happened but it seems its messed up.

**Comment by sparerd** [comment]
Looks like the issue occurs if you have the save-on-close setting enabled. The saving mechanism has no concept of safety and will overwrite the valid-but-still-encrypted connections file with whatever is in the tree (which is nothing).

This seems like a somewhat serious bug, in that it can wipe out connections files under certain situations.

**Comment by Silent-Joker** [comment]
ok that is why i edited the post.  I thought that was what happened but it was a busy day.

That also answered the question i was wondering where that "Putty Saved Sessions" came from :)  Taught me a lesson on not backing up my file after major additions to connections.  

So than would the plan be to keep the password file as the file that should open at launch still even if you hit cancel; unless you open another file of course?

**Comment by sparerd** [comment]
Correct, though there is some complexity here around that. There are a few use cases:

After cancelling the password prompt, you could:
1. Exit mRemoteNG
1. Try to save the current (empty) connections tree.
1. File -> Open and try to load the file again

Case 1: we should NOT save anything (even if the save-on-close option is set) and try to load the same file on next start up.

Case 2: force all normal "save" actions to be "save-as" actions instead to prevent accidental overwrites.

Case 3: nothing special here, just try to reload the file.

Case 1 is very serious and should be fixed immediately, case 2 is somewhat important and would be a good safety measure, and I think case 3 is already working as expected.

**Comment by sparerd** [comment]
Resolved case 1 with a minor change. There are still some improvements I'd like to make, though the worst part of the issue is resolved.

**Comment by Silent-Joker** [comment]
I think option 3. File -> Open would be easier if there was a recently opened list under the open menu.  This way if you forget like i did for a few hours on where it was saved you can now just click the recent opened one :)

Thats more extra topping though and not needed immediately...

Glad case 1 is fixed though :)

**Comment by Kvarkas** [comment]
not relevant for new version
